examples/keygen/keyload.c

```diff
@@ -0,0 +1,159 @@
+#include <examples/tpm_test.h>
+
+#include <stdio.h>
+
+
+/******************************************************************************/
+/* --- BEGIN TPM Key Load Example -- */
+/******************************************************************************/
+
+int TPM2_Keyload_Example(void* userCtx, int argc, char *argv[])
+{
+    int rc;
+    WOLFTPM2_DEV dev;
+    WOLFTPM2_KEY storage; /* SRK */
+    WOLFTPM2_KEYBLOB newKey;
+    TPMS_AUTH_COMMAND session[MAX_SESSION_NUM];
+#if !defined(WOLFTPM2_NO_WOLFCRYPT) && !defined(NO_FILESYSTEM)
+    XFILE f;
+    const char* inputFile = "keyblob.bin";
+
+    if (argc >= 2) {
+        inputFile = argv[1];
+    }
+#else
+    (void)argc;
+    (void)argv;
+#endif
+
+    XMEMSET(session, 0, sizeof(session));
+    XMEMSET(&storage, 0, sizeof(storage));
+    XMEMSET(&newKey, 0, sizeof(newKey));
+
+    printf("TPM2.0 Key load example\n");
+    rc = wolfTPM2_Init(&dev, TPM2_IoCb, userCtx);
+    if (rc != TPM_RC_SUCCESS) {
+        printf("\nwolfTPM2_Init failed\n");
+        goto exit;
+    }
+    printf("wolfTPM2_Init: success\n\n");
+
+
+    /* Define the default session auth that has NULL password */
+    session[0].sessionHandle = TPM_RS_PW;
+    session[0].auth.size = 0;
+    TPM2_SetSessionAuth(session);
+
+    /* See if SRK already exists */
+    rc = wolfTPM2_ReadPublicKey(&dev, &storage, TPM2_DEMO_STORAGE_KEY_HANDLE);
+    if (rc != 0) {
+        printf("Loading SRK: Storage failed 0x%x: %s\n", rc,
+            TPM2_GetRCString(rc));
+        goto exit;
+    }
+    printf("Loading SRK: Storage 0x%x (%d bytes)\n",
+        (word32)storage.handle.hndl, storage.pub.size);
+
+    /* set session for authorization of the storage key */
+    session[0].auth.size = sizeof(gStorageKeyAuth)-1;
+    XMEMCPY(session[0].auth.buffer, gStorageKeyAuth, session[0].auth.size);
+
+    /* Load encrypted key from the disk */
+#if !defined(WOLFTPM2_NO_WOLFCRYPT) && !defined(NO_FILESYSTEM)
+    f = XFOPEN(inputFile, "rb");
+    if (f != XBADFILE) {
+        size_t fileSz;
+        XFSEEK(f, 0, XSEEK_END);
+        fileSz = XFTELL(f);
+        XREWIND(f);
+        if (fileSz > sizeof(newKey.priv) + sizeof(newKey.pub)) {
+            printf("File size check failed\n");
+            rc = BUFFER_E; goto exit;
+        }
+        printf("Reading %d bytes from %s\n", (int)fileSz, inputFile);
+        
+        XFREAD(&newKey.pub, 1, sizeof(newKey.pub), f);
+        if (fileSz > sizeof(newKey.pub)) {
+            fileSz -= sizeof(newKey.pub);
+            XFREAD(&newKey.priv, 1, fileSz, f);
+        }
+        XFCLOSE(f);
+
+        /* sanity check the sizes */
+        if (newKey.pub.size != sizeof(newKey.pub) || newKey.priv.size > sizeof(newKey.priv.buffer)) {
+            printf("Struct size check failed (pub %d, priv %d)\n", newKey.pub.size, newKey.priv.size);
+            rc = BUFFER_E; goto exit;
+        }
+    }
+    else {
+        printf("File %s not found!\n", inputFile);
+        goto exit;
```

Comment:
nit: if file doesn't exist, rc is 0 and the process exits with 0. I'd probably also exit with non-0 on loading blob not supported also.